app/src/main/java/de/westnordost/streetcomplete/quests/building_type/AddBuildingType.kt

```diff
@@ -36,8 +37,9 @@ class AddBuildingType : OsmFilterQuestType<String>() {
             val manMade = answer.split("=")[1]
             changes.delete("building")
             changes.add("man_made", manMade)
-        } else {
+        } else if (answer == "historic" || answer == "abandoned" || answer == "ruins")
+            changes.add(answer, "yes")
```

Comment:
Hmm, this will crash the app whenever there was a previous value for either of `history`, `abandoned` or `ruins` that was not `yes`.
```suggestion
            changes.addOrModify(answer, "yes")
```